graphql_compiler/query_planning/typedefs.py

```diff
@@ -0,0 +1,310 @@
+            return False
+
+    # All keys and values match so return True.
+    return True
+
+
+def _deserialize_independent_query_plan_field(dict_value: Dict[str, Any]) -> "IndependentQueryPlan":
+    """Deserialize the dict representation of IndependentQueryPlan."""
+    # Note: there will be more types of IndependentQueryPlans that will require different
+    # deserialization shortly.
+    return SimpleExecute.from_dict(dict_value)
+
+
+# ############
+# Public API #
+# ############
+
+
+@unique
+class BackendType(Enum):
+    # N.B.: The values of the enums are the "human-friendly" display names. Since they are shown
+    #       to humans, they are subject to change if said humans find a friendlier name.
+    #       Don't assume they are immutable!
+    cypher = "Cypher"
+    gremlin = "Gremlin"
+    interpreter = "InterpreterAdapter"
+    match = "Match"
+    mssql = "MSSQL"
+    postgresql = "PostgreSQL"
+
+
+@dataclass(init=True, repr=True, eq=True, frozen=True)
+class ProviderMetadata(DataClassJsonMixin):
+    """Metadata about the provider."""
+
+    # Name of the type of provider.
+    backend_type: BackendType
+
+    # Whether this backend requires MSSQL fold postprocessing for folded outputs.
+    requires_fold_postprocessing: bool
+
+
+@dataclass(init=True, repr=True, eq=True, frozen=True)
+class QueryPlanNode(DataClassJsonMixin, metaclass=ABCMeta):
+    """Abstract query plan node. May or may not contain other nodes, depending on its type."""
+
+    # Unique ID of the plan node.
+    # Note: do not compare "uuid" values to determine whether query plan nodes are equal, since two
+    # plans with different identifiers might still be semantically equivalent and therefore equal.
+    uuid: str = field(compare=False)
+
+
+@dataclass(init=True, repr=True, eq=False, frozen=True)
+class SimpleExecute(QueryPlanNode):
+    """Just give the specified query and args to the provider, it'll execute it for you as-is."""
+
+    provider_id: str
+    provider_metadata: ProviderMetadata
+    query: str  # in whatever query language the provider will accept (not necessarily GraphQL)
+    arguments: Dict[str, Any]
+
+    # Input and output metadata of the query.
+    output_metadata: Dict[str, OutputMetadata] = field(
+        metadata=config(
+            encoder=_serialize_output_metadata_field, decoder=_deserialize_output_metadata_field
+        )
+    )
+    input_metadata: Dict[str, QueryArgumentGraphQLType] = field(
+        metadata=config(
+            encoder=_serialize_input_metadata_field, decoder=_deserialize_input_metadata_field
+        )
+    )
+
+    def __eq__(self, other: Any) -> bool:
+        """Check equality between an object and this SimpleExecute."""
+        if not isinstance(other, SimpleExecute):
+            return False
+
+        # Perform special check for input_metadata since GraphQLTypes don't have equality, and
+        # check all other fields in a straight forward manner.
```

Comment:
Perhaps one word instead of two?
```suggestion
        # check all other fields in a straightforward manner.
```